File: devtools/ui-tools/update-dialog-style.js
/**
 * סקריפט Node.js לעדכון הדיאלוג - רק header + buttons
 * עדכון זהיר לפי כללי הפרויקט
 */

const fs = require('fs');
const path = require('path');

function updateDialogFile() {
  console.log('🚀 מתחיל עדכון...\n');

  const filePath = path.join(__dirname, 'js', 'modules', 'case-creation', 'case-creation-dialog.js');

  // קריאת הקובץ
  let content = fs.readFileSync(filePath, 'utf-8');

  // ✅ שינוי 1: Overlay div
  content = content.replace(
    /<div id="modernCaseDialog" style="[^"]*">/,
    '<div id="modernCaseDialog" class="case-dialog-overlay">'
  );

  // ✅ שינוי 2: Container div
  content = content.replace(
    /(<div id="modernCaseDialog"[^>]*>\s*)<div style="\s*background: white;[\s\S]*?animation: slideUp[^"]*">/,
    '$1<div class="case-dialog-container">'
  );

  // ✅ שינוי 3: Header section
  content = content.replace(
    /<!-- Header -->\s*<div style="[\s\S]*?linear-gradient[\s\S]*?">/,
    '<!-- Header -->\n            <div class="case-dialog-header">'
  );

  // ✅ שינוי 4: Header content div
  content = content.replace(
    /(<div class="case-dialog-header">\s*)<div style="display: flex;[^"]*">/,
    '$1<div class="case-dialog-header-content">'
  );

  // ✅ שינוי 5: הסרת style מהאייקון בheader
  content = content.replace(
    /<i class="fas fa-folder-plus" style="[^"]*">/,
    '<i class="fas fa-folder-plus">'
  );

  // ✅ שינוי 6: הסרת style מה-h2
  content = content.replace(
    /<h2 style="[^"]*">תיק חדש<\/h2>/,
    '<h2>תיק חדש</h2>'
  );

  // ✅ שינוי 7: כפתור סגירה
  content = content.replace(
    /<button id="modernCaseDialog_close" style="[\s\S]*?">/,
    '<button id="modernCaseDialog_close" class="case-dialog-close">'
  );

  // ✅ שינוי 8: Content div
  content = content.replace(
    /<!-- Content -->\s*<div style="padding:[^"]*">/,
    '<!-- Content -->\n            <div class="case-dialog-content">'
  );

  // ✅ שינוי 9: Buttons container
  content = content.replace(
    /<div style="\s*display: flex;[\s\S]*?justify-content: flex-end;[\s\S]*?">/,
    '<div class="case-dialog-actions">'
  );

  // ✅ שינוי 10: כפתור ביטול
  content = content.replace(
    /<button type="button" id="modernCaseDialog_cancel" style="[\s\S]*?">/,
    '<button type="button" id="modernCaseDialog_cancel" class="btn btn-secondary">'
  );

  // ✅ שינוי 11: כפתור שמירה
  content = content.replace(
    /<button type="submit" style="[\s\S]*?linear-gradient[\s\S]*?">/,
    '<button type="submit" class="btn btn-primary">'
  );

  // ✅ שינוי 12: הסרת style מאייקון השמירה
  content = content.replace(
    /<i class="fas fa-save" style="[^"]*">/,
    '<i class="fas fa-save">'
  );

  // ✅ שינוי 13: הסרת <style> בסוף
  content = content.replace(
    /<style>[\s\S]*?@keyframes[\s\S]*?<\/style>/,
    '<!-- Animations moved to case-creation-dialog.css -->'
  );

  // שמירת הקובץ
  fs.writeFileSync(filePath, content, 'utf-8');

  console.log('✅ הקובץ עודכן בהצלחה!');
  console.log(`📁 ${filePath}\n`);
  console.log('🔍 שינויים שבוצעו:');
  console.log('  1. ✅ Overlay div → class="case-dialog-overlay"');
  console.log('  2. ✅ Container div → class="case-dialog-container"');
  console.log('  3. ✅ Header section → class="case-dialog-header"');
  console.log('  4. ✅ Header content → class="case-dialog-header-content"');
  console.log('  5. ✅ Close button → class="case-dialog-close"');
  console.log('  6. ✅ Content div → class="case-dialog-content"');
  console.log('  7. ✅ Buttons container → class="case-dialog-actions"');
  console.log('  8. ✅ Cancel button → class="btn btn-secondary"');
  console.log('  9. ✅ Submit button → class="btn btn-primary"');
  console.log(' 10. ✅ הסרת inline styles מאייקונים');
  console.log(' 11. ✅ הסרת <style> tags (אנימציות)\n');
  console.log('🎨 עכשיו הדיאלוג משתמש בסטייל Linear/Vercel!');
  console.log('\n📝 לבדיקה:');
  console.log('   1. פתח את הדפדפן');
  console.log('   2. לחץ על "תיק חדש"');
  console.log('   3. בדוק את העיצוב החדש');
}

try {
  updateDialogFile();
} catch (error) {
  console.error('❌ שגיאה:', error.message);
  console.error(error.stack);
  process.exit(1);
}

File: devtools/ui-tools/update-dialog-style.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
סקריפט לעדכון הדיאלוג - רק header + buttons
עדכון זהיר לפי כללי הפרויקט
"""

import re

def update_dialog_file():
    """עדכון הקובץ case-creation-dialog.js"""

    file_path = r'js\modules\case-creation\case-creation-dialog.js'

    # קריאת הקובץ
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # ✅ שינוי 1: Overlay div
    content = re.sub(
        r'<div id="modernCaseDialog" style="[^"]*">',
        '<div id="modernCaseDialog" class="case-dialog-overlay">',
        content
    )

    # ✅ שינוי 2: Container div (הראשון אחרי ה-overlay)
    # מחפש את הדיב הראשון בתוך ה-overlay
    content = re.sub(
        r'(<div id="modernCaseDialog"[^>]*>\s*)<div style="\s*background: white;[^"]*">',
        r'\1<div class="case-dialog-container">',
        content
    )

    # ✅ שינוי 3: Header section - הסרת הגרדיאנט
    content = re.sub(
        r'<!-- Header -->\s*<div style="[^"]*background:[^"]*linear-gradient[^"]*">',
        '<!-- Header -->\n            <div class="case-dialog-header">',
        content
    )

    # ✅ שינוי 4: Header content div
    content = re.sub(
        r'(<div class="case-dialog-header">\s*)<div style="display: flex;[^"]*">',
        r'\1<div class="case-dialog-header-content">',
        content
    )

    # ✅ שינוי 5: הסרת style מהאייקון בheader
    content = re.sub(
        r'<i class="fas fa-folder-plus" style="[^"]*">',
        '<i class="fas fa-folder-plus">',
        content
    )

    # ✅ שינוי 6: הסרת style מה-h2 בheader
    content = re.sub(
        r'<h2 style="[^"]*">תיק חדש</h2>',
        '<h2>תיק חדש</h2>',
        content
    )

    # ✅ שינוי 7: כפתור סגירה
    content = re.sub(
        r'<button id="modernCaseDialog_close" style="[^"]*">',
        '<button id="modernCaseDialog_close" class="case-dialog-close">',
        content
    )

    # ✅ שינוי 8: Content div
    content = re.sub(
        r'<!-- Content -->\s*<div style="padding:[^"]*">',
        '<!-- Content -->\n            <div class="case-dialog-content">',
        content
    )

    # ✅ שינוי 9: Buttons container - מציאת הדיב עם justify-content: flex-end
    content = re.sub(
        r'<div style="\s*display: flex;\s*gap: 12px;\s*justify-content: flex-end;[^"]*">',
        '<div class="case-dialog-actions">',
        content
    )

    # ✅ שינוי 10: כפתור ביטול
    content = re.sub(
        r'<button type="button" id="modernCaseDialog_cancel" style="[^"]*">',
        '<button type="button" id="modernCaseDialog_cancel" class="btn btn-secondary">',
        content
    )

    # ✅ שינוי 11: כפתור שמירה (עם gradient)
    content = re.sub(
        r'<button type="submit" style="[^"]*background:[^"]*linear-gradient[^"]*">',
        '<button type="submit" class="btn btn-primary">',
        content
    )

    # ✅ שינוי 12: הסרת style מאייקון השמירה
    content = re.sub(
        r'<i class="fas fa-save" style="[^"]*">',
        '<i class="fas fa-save">',
        content
    )

    # ✅ שינוי 13: הסרת <style> בסוף (אנימציות)
    content = re.sub(
        r'<style>\s*@keyframes fadeIn[^<]*</style>',
        '<!-- Animations moved to case-creation-dialog.css -->',
        content,
        flags=re.DOTALL
    )

    # שמירת הקובץ המעודכן
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)

    print('✅ הקובץ עודכן בהצלחה!')
    print(f'📁 {file_path}')
    print('\n🔍 שינויים שבוצעו:')
    print('  1. ✅ Overlay div → class="case-dialog-overlay"')
    print('  2. ✅ Container div → class="case-dialog-container"')
    print('  3. ✅ Header section → class="case-dialog-header"')
    print('  4. ✅ Header content → class="case-dialog-header-content"')
    print('  5. ✅ Close button → class="case-dialog-close"')
    print('  6. ✅ Content div → class="case-dialog-content"')
    print('  7. ✅ Buttons container → class="case-dialog-actions"')
    print('  8. ✅ Cancel button → class="btn btn-secondary"')
    print('  9. ✅ Submit button → class="btn btn-primary"')
    print(' 10. ✅ הסרת inline styles מאייקונים')
    print(' 11. ✅ הסרת <style> tags (אנימציות)')
    print('\n🎨 עכשיו הדיאלוג משתמש בסטייל Linear/Vercel!')

if __name__ == '__main__':
    try:
        update_dialog_file()
    except Exception as e:
        print(f'❌ שגיאה: {e}')
        import traceback
        traceback.print_exc()
